### PR TITLE
Replaced json strings with DTOs

### PR DESCRIPTION
**Pull request by SankethBK**

### CODE REVIEW

**Comment by SankethBK** [comment]
@guyluz11 I changed SwitchState to DeviceActions.on.toString()... still not able to see the on/off buttons. Can you please help

File: lib/infrastructure/hub_client/hub_client_demo.dart [hunk 1]
```diff
@@ -81,62 +149,134 @@ class HubClientDemo {
...
 
-    const String allRemoteCommandsBlinds =
-        '{"id":"96386cd0-419b-11ed-8246-e17a279f4d89","vendorUniqueId":"63b298","defaultName":"Guy Blinds","deviceStateGRPC":"ack","senderDeviceOs":"switcher","senderDeviceModel":"SwitcherDevicesTypes.switcherRunner","senderId":"94edb282-434a-11ed-bd96-f573a00b65aa","blindsSwitchState":"stop","deviceTypes":"blinds","compUuid":"34asdfrsd23gggg","deviceVendor":"switcherSmartHome","deviceDtoClass":"GenericBlindsDeviceDtos","stateMassage":"Hello World"}';
+    final GenericBlindsDE allRemoteCommandsBlinds = GenericBlindsDE(
+      uniqueId:
+          CoreUniqueId.fromUniqueString("96386cd0-419b-11ed-8246-e17a279f4d89"),
+      vendorUniqueId: VendorUniqueId.fromUniqueString("63b298"),
+      deviceVendor: DeviceVendor("switcherSmartHome"),
+      defaultName: DeviceDefaultName("Guy Blinds"),
+      deviceStateGRPC: DeviceState("ack"),
```
Comment (guyluz11) [inline lib/infrastructure/hub_client/hub_client_demo.dart:158]: This value should come from the state veritable that exists in the `.grpc` file of the hub.

`DeviceStateGRPC`

File: lib/infrastructure/hub_client/hub_client_demo.dart [hunk 2]
```diff
@@ -81,62 +149,134 @@ class HubClientDemo {
...
+      stateMassage: DeviceStateMassage("Hello World"),
+      senderDeviceOs: DeviceSenderDeviceOs("switcher"),
+      senderDeviceModel:
+          DeviceSenderDeviceModel("SwitcherDevicesTypes.switcherRunner"),
+      senderId: DeviceSenderId.fromUniqueString(
+          "94edb282-434a-11ed-bd96-f573a00b65aa"),
+      compUuid: DeviceCompUuid("34asdfrsd23gggg"),
+      blindsSwitchState: GenericBlindsSwitchState("stop"),
```
Comment (guyluz11) [inline lib/infrastructure/hub_client/hub_client_demo.dart:166]: This value should come from the device state veritable that exists in the .grpc file of the hub.

 `DeviceActions.`

File: lib/infrastructure/hub_client/hub_client_demo.dart [hunk 3]
```diff
@@ -81,62 +149,134 @@ class HubClientDemo {
...
 
-    const String allRemoteCommandsButton =
-        '{"id":"c90137f2-419b-11ed-8246-e17a279f4d89","vendorUniqueId":"67600068c44f33f5b515","defaultName":"Guy Ceiling Button","deviceStateGRPC":"ack","senderDeviceOs":"tuya_smart","senderDeviceModel":"Cloud","senderId":"90d20fc3-434a-11ed-bd96-f573a00b65aa","switchState":"false","deviceTypes":"switch","compUuid":"34asdfrsd23gggg","deviceVendor":"tuyaSmart","deviceDtoClass":"GenericSwitchDeviceDtos","stateMassage":"Hello World"}';
+    final GenericBlindsDE allRemoteCommandsButton = GenericBlindsDE(
+      uniqueId:
+          CoreUniqueId.fromUniqueString("c90137f2-419b-11ed-8246-e17a279f4d89"),
+      vendorUniqueId: VendorUniqueId.fromUniqueString("67600068c44f33f5b515"),
+      deviceVendor: DeviceVendor("tuyaSmart"),
+      defaultName: DeviceDefaultName("Guy Ceiling Button"),
+      deviceStateGRPC: DeviceState("ack"),
```
Comment (guyluz11) [inline lib/infrastructure/hub_client/hub_client_demo.dart:186]: Same

File: lib/infrastructure/hub_client/hub_client_demo.dart [hunk 4]
```diff
@@ -81,62 +149,134 @@ class HubClientDemo {
...
+          DeviceHelper.convertDomainToJsonString(allRemoteCommandsBlinds),
     );
 
     devicesList.add(requestsAndStatusFromHubBlinds);
 
     /// Guy Ceiling Button
 
-    const String allRemoteCommandsButton =
-        '{"id":"c90137f2-419b-11ed-8246-e17a279f4d89","vendorUniqueId":"67600068c44f33f5b515","defaultName":"Guy Ceiling Button","deviceStateGRPC":"ack","senderDeviceOs":"tuya_smart","senderDeviceModel":"Cloud","senderId":"90d20fc3-434a-11ed-bd96-f573a00b65aa","switchState":"false","deviceTypes":"switch","compUuid":"34asdfrsd23gggg","deviceVendor":"tuyaSmart","deviceDtoClass":"GenericSwitchDeviceDtos","stateMassage":"Hello World"}';
+    final GenericBlindsDE allRemoteCommandsButton = GenericBlindsDE(
```
Comment (guyluz11) [inline lib/infrastructure/hub_client/hub_client_demo.dart:180]: The device type was a switch so this needs to change info `GenericSwitchDE`

File: lib/infrastructure/hub_client/hub_client_demo.dart [hunk 5]
```diff
@@ -81,62 +149,134 @@ class HubClientDemo {
...
+      allRemoteCommands:
+          DeviceHelper.convertDomainToJsonString(allRemoteCommandsButton),
     );
     devicesList.add(requestsAndStatusFromHubButton);
 
     /// Guy Button 2
 
-    const String allRemoteCommandsButton2 =
-        '{"id":"a31523m6-463s-32ge-7426-g33c642r7m25","vendorUniqueId":"67600068c44f33f5b515","defaultName":"Guy Bedroom Button","deviceStateGRPC":"ack","senderDeviceOs":"tuya_smart","senderDeviceModel":"Cloud","senderId":"90d20fc3-434a-11ed-bd96-f573a00b65aa","switchState":"false","deviceTypes":"switch","compUuid":"34asdfrsd23gggg","deviceVendor":"tuyaSmart","deviceDtoClass":"GenericSwitchDeviceDtos","stateMassage":"Hello World"}';
+    final GenericBlindsDE allRemoteCommandsButton2 = GenericBlindsDE(
```
Comment (guyluz11) [inline lib/infrastructure/hub_client/hub_client_demo.dart:206]: Another switch DE and not blinds

File: lib/infrastructure/hub_client/hub_client_demo.dart [hunk 6]
```diff
@@ -81,62 +149,134 @@ class HubClientDemo {
...
+      allRemoteCommands:
+          DeviceHelper.convertDomainToJsonString(allRemoteCommandsButton2),
     );
     devicesList.add(requestsAndStatusFromHubButton2);
 
     /// Boiler
 
-    const String allRemoteCommandsBoiler =
-        '{"id":"65d84b10-434d-11ed-817a-7d350fb52f91","vendorUniqueId":"65fd2f","defaultName":"Boiler","deviceStateGRPC":"ack","senderDeviceOs":"switcher","senderDeviceModel":"SwitcherDevicesTypes.switcherV2Esp","senderId":"65d84b11-434d-11ed-817a-7d350fb52f91","boilerSwitchState":"off","deviceTypes":"boiler","compUuid":"34asdfrsd23gggg","deviceVendor":"switcherSmartHome","deviceDtoClass":"GenericBoilerDeviceDtos","stateMassage":"Hello World"}';
+    final GenericBlindsDE allRemoteCommandsBoiler = GenericBlindsDE(
```
Comment (guyluz11) [inline lib/infrastructure/hub_client/hub_client_demo.dart:232]: Needs to be changed into boilerDE

File: lib/infrastructure/hub_client/hub_client_demo.dart [hunk 7]
```diff
@@ -81,62 +149,134 @@ class HubClientDemo {
...
+          DeviceHelper.convertDomainToJsonString(allRemoteCommandsBoiler),
     );
     devicesList.add(requestsAndStatusFromHubBoiler);
 
     /// RGBLight
 
-    const String allRemoteCommandsRgbLight =
-        '{"id":"7189ed76-4351-11ed-b249-63fd7e165c16","vendorUniqueId":"bf4bccdaf68347e31dhiut","defaultName":"Dining Room Ceiling","deviceStateGRPC":"ack","senderDeviceOs":"tuya_smart","senderDeviceModel":"1SE","senderId":"7189ed77-4351-11ed-b249-63fd7e165c16","deviceTypes":"rgbwLights","compUuid":"34asdfrsd23gggg","deviceVendor":"tuyaSmart","lightSwitchState":"on","lightColorTemperature":"8129","lightBrightness":"1000","lightColorAlpha":"1.0","lightColorHue":"0.0","lightColorSaturation":"1.0","lightColorValue":"1.0","deviceDtoClass":"GenericRgbwLightDeviceDtos","stateMassage":"Hello World"}';
+    final GenericBlindsDE allRemoteCommandsRgbLight = GenericBlindsDE(
+      uniqueId:
```
Comment (guyluz11) [inline lib/infrastructure/hub_client/hub_client_demo.dart:260]: Needs to be changed into rgbwlightsDE

File: lib/infrastructure/hub_client/hub_client_demo.dart [hunk 8]
```diff
@@ -150,12 +290,31 @@ class HubClientDemo {
...
     /// Study Scene
 
-    const String allRemoteCommandsSceneOpen =
-        '''{"uniqueId":"01bd6880-419b-11ed-af10-f75196d26a5e","name":"Study Scene","backgroundColor":"0xFFFF9800","deviceStateGRPC":"waitingInComp","senderDeviceOs":null,"senderDeviceModel":null,"senderId":null,"compUuid":null,"stateMassage":null,"automationString":"[]","nodeRedFlowId":"6b63068c3c970eeb","firstNodeId":"3f1a9440-41a1-11ed-8b6a-5f136f4fe196","iconCodePoint":null,"image":null,"lastDateOfExecute":null}''';
+    final SceneCbjEntity allRemoteCommandsSceneOpen = SceneCbjEntity(
+      uniqueId:
+          UniqueId.fromUniqueString("01bd6880-419b-11ed-af10-f75196d26a5e"),
+      name: SceneCbjName("Study Scene"),
+      backgroundColor: SceneCbjBackgroundColor("0xFFFF9800"),
+      nodeRedFlowId: SceneCbjNodeRedFlowId("waitingInComp"),
```
Comment (guyluz11) [inline lib/infrastructure/hub_client/hub_client_demo.dart:298]: Value got inserted incorrectly

**Review by guyluz11 — CHANGES_REQUESTED** [review]
Please fix all the fields that I have marked to the correct types.

There are more changes that need to be looked over, some of the states are in the incorrect fields.

File: lib/infrastructure/hub_client/hub_client_demo.dart (hunk 1 above)
Comment (SankethBK) [reply lib/infrastructure/hub_client/hub_client_demo.dart:164]: Hi, not able to understand how the to use values defined in grpc files, i checked in hub project. Can you please help

File: lib/infrastructure/hub_client/hub_client_demo.dart [hunk 9]
```diff
@@ -150,12 +303,31 @@ class HubClientDemo {
...
-    const String allRemoteCommandsSceneOpen =
-        '''{"uniqueId":"01bd6880-419b-11ed-af10-f75196d26a5e","name":"Study Scene","backgroundColor":"0xFFFF9800","deviceStateGRPC":"waitingInComp","senderDeviceOs":null,"senderDeviceModel":null,"senderId":null,"compUuid":null,"stateMassage":null,"automationString":"[]","nodeRedFlowId":"6b63068c3c970eeb","firstNodeId":"3f1a9440-41a1-11ed-8b6a-5f136f4fe196","iconCodePoint":null,"image":null,"lastDateOfExecute":null}''';
+    final SceneCbjEntity allRemoteCommandsSceneOpen = SceneCbjEntity(
+      uniqueId:
+          UniqueId.fromUniqueString("01bd6880-419b-11ed-af10-f75196d26a5e"),
+      name: SceneCbjName("Study Scene"),
+      backgroundColor: SceneCbjBackgroundColor("0xFFFF9800"),
+      nodeRedFlowId: SceneCbjNodeRedFlowId("6b63068c3c970eeb"),
+      automationString: SceneCbjAutomationString("[]"),
+      firstNodeId: SceneCbjFirstNodeId("6b63068c3c970eeb"),
```
Comment (guyluz11) [inline lib/infrastructure/hub_client/hub_client_demo.dart:313]: Incorrect value

**Review by guyluz11 — CHANGES_REQUESTED** [review]
Here you go.
User ack like this `DeviceStateGRPC.ack.toString()`

File: lib/infrastructure/hub_client/hub_client_demo.dart [hunk 10]
```diff
@@ -81,62 +155,141 @@ class HubClientDemo {
...
 
-    const String allRemoteCommandsRgbLight =
-        '{"id":"7189ed76-4351-11ed-b249-63fd7e165c16","vendorUniqueId":"bf4bccdaf68347e31dhiut","defaultName":"Dining Room Ceiling","deviceStateGRPC":"ack","senderDeviceOs":"tuya_smart","senderDeviceModel":"1SE","senderId":"7189ed77-4351-11ed-b249-63fd7e165c16","deviceTypes":"rgbwLights","compUuid":"34asdfrsd23gggg","deviceVendor":"tuyaSmart","lightSwitchState":"on","lightColorTemperature":"8129","lightBrightness":"1000","lightColorAlpha":"1.0","lightColorHue":"0.0","lightColorSaturation":"1.0","lightColorValue":"1.0","deviceDtoClass":"GenericRgbwLightDeviceDtos","stateMassage":"Hello World"}';
+    final GenericRgbwLightDE allRemoteCommandsRgbLight = GenericRgbwLightDE(
+      uniqueId:
+          CoreUniqueId.fromUniqueString("7189ed76-4351-11ed-b249-63fd7e165c16"),
+      vendorUniqueId: VendorUniqueId.fromUniqueString("bf4bccdaf68347e31dhiut"),
+      deviceVendor: DeviceVendor("tuyaSmart"),
+      defaultName: DeviceDefaultName("Dining Room Ceiling"),
+      deviceStateGRPC: DeviceState("ack"),
```
[the remark this answers is not in the document]
Comment (SankethBK) [reply lib/infrastructure/hub_client/hub_client_demo.dart:271]: I couldn't find a class named   `DeviceStateGRPC ` 
But in this file `device_entity_abstract.dart`, this value for a similar field was like this
`        deviceStateGRPC: DeviceState(DeviceTypes.typeNotSupported.toString()),
`

File: lib/infrastructure/hub_client/hub_client_demo.dart (hunk 7 above)
Comment (SankethBK) [reply lib/infrastructure/hub_client/hub_client_demo.dart:266]: i have changed it, it's not updating here

File: lib/infrastructure/hub_client/hub_client_demo.dart (hunk 10 above)
[the remark this answers is not in the document]
Comment (guyluz11) [reply lib/infrastructure/hub_client/hub_client_demo.dart:278]: The location of `DeviceStateGRPC` is from grpc generated files 
`cbj_app/lib/infrastructure/core/gen/cbj_hub_server/protoc_as_dart/cbj_hub_server.pbenum.dart`

**Review by guyluz11 — CHANGES_REQUESTED** [review]

**Comment by guyluz11** [comment]
Are you able to turn on off buttons now?

> @guyluz11 I changed SwitchState to DeviceActions.on.toString()... still not able to see the on/off buttons. Can you please help

**Comment by SankethBK** [comment]
> Are you able to turn on off buttons now?
> 
> > @guyluz11 I changed SwitchState to DeviceActions.on.toString()... still not able to see the on/off buttons. Can you please help

Yes, they're working now

**Comment by guyluz11** [comment]
Thanks for the pr and a happy Hacktoberfest 😁

**Pull request merged**